test/utils.sh

```diff
@@ -227,3 +227,88 @@ get_image_manifests() {
   echo "${image_manifests}"
 
 }
+
+# This function will be used by tekton tasks in build-definitions
+# It returns a list of unreleased bundles in the FBC fragment by comparing it with
+# the corresponding production index image
+get_unreleased_bundle() {
+  # FBC fragment containing the unreleased bundle
+  local FBC_FRAGMENT=""
+
+  get_unreleased_bundle_usage()
+  {
+    echo "
+  get_unreleased_bundle  -i FBC_FRAGMENT
+  " >&2
+    exit 2
+  }
+
+  local OPTIND opt
+  while getopts "i:" opt; do
+      case "${opt}" in
+          i)
+              FBC_FRAGMENT="${OPTARG}" ;;
+          *)
+              get_unreleased_bundle_usage ;;
+      esac
+  done
+  shift $((OPTIND-1))
+
+  if [ -z "$FBC_FRAGMENT" ]; then
+    echo "Missing parameter FBC_FRAGMENT" >&2
+    exit 2
+  fi
+
+  local INDEX_IMAGE="registry.redhat.io/redhat/redhat-operator-index"
+
+  #Get target ocp version from the fragment
+  if ! ocp_version=$(skopeo inspect --no-tags --raw docker://"$FBC_FRAGMENT"); then
+    echo "The raw image inspect command failed" >&2
+    exit 1
+  fi
+  ocp_version=$(echo "$ocp_version" |  jq -r '.annotations."org.opencontainers.image.base.name"' | cut -d: -f2)
+
+  # Decide which OPM version to use for rendering the image
+  local opm_version
+  if [[ "$(printf '%s\n' "$ocp_version" "v4.17" | sort -V | head -n1)" == "v4.17" ]]; then
+    opm_version="opm-v1.40.0"
+  else
+    opm_version="opm"
+  fi
+
+
+  # Run opm render on the FBC fragment to extract package name and unique bundles
+  local render_out_fbc unique_bundles_fbc package_name
+  if ! render_out_fbc=$($opm_version render "$FBC_FRAGMENT"); then
+    echo "Could not render FBC Fragment" >&2
+    exit 1
+  fi
+  package_name=$(echo "$render_out_fbc" | jq -r 'select(.schema == "olm.package") | .name')
```

Comment:
We also do this in the release pipeline: https://github.com/konflux-ci/release-service-catalog/blob/development/tasks/check-fbc-packages/check-fbc-packages.yaml#L44

Reply:
Somewhat tangential, but are we sure we want to make assumptions about fragments containing exactly one package?

If so, maybe that's fine for now, but would there be a way to centralize logic around that such that if there are future use cases around "no packages" (i.e. maybe some sort of catalog-wide metadata) or "multiple packages" (i.e. maybe a "package group") in a fragment, it isn't a game of whack-a-mole to find them.

Reply:
Oh, I didn't catch that. We shouldn't assume only one package.

Reply:
I don't think that we require that in the release catalog which has the same implementation though. So maybe this will actually return a list? Either way, this code wouldn't handle that scenario.

Reply:
@arewm Do we want this code to handle the multiple packages scenario?

Reply:
Yes, all other parts of the workflow should support building and releasing multiple packages within a single fragment.

Reply:
The code now supports multiple packages in a single fragment. Please let me know what you think. thanks in advance!